src/test/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -0,0 +1,162 @@
+
+package nearsoft.academy.bigdata.recommendation;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+
+import org.apache.mahout.cf.taste.impl.model.GenericPreference;
+import org.apache.mahout.cf.taste.common.TasteException;
+import org.apache.mahout.cf.taste.impl.common.*;
+import org.apache.mahout.cf.taste.impl.model.GenericUserPreferenceArray;
+import org.apache.mahout.cf.taste.impl.model.GenericDataModel;
+import org.apache.mahout.cf.taste.impl.neighborhood.ThresholdUserNeighborhood;
+import org.apache.mahout.cf.taste.impl.recommender.GenericUserBasedRecommender;
+import org.apache.mahout.cf.taste.model.PreferenceArray;
+import org.apache.mahout.cf.taste.similarity.UserSimilarity;
+import org.apache.mahout.cf.taste.impl.similarity.PearsonCorrelationSimilarity;
+import org.apache.mahout.cf.taste.neighborhood.UserNeighborhood;
+import org.apache.mahout.cf.taste.recommender.RecommendedItem;
+
+public class MovieRecommender {
+
+    private FastByIDMap<PreferenceArray> usersPreferences;
+    private HashMap<String, Long> userIDs;
+    private BiMap<String, Long> productIDsBi;
+
+    private long userCount;
+    private long productCount;
+    private long reviewCount;
+    private GenericDataModel model;
+    private GenericUserBasedRecommender recommender;
+
+    public MovieRecommender(String filePath) throws TasteException {
+
+        productIDsBi = HashBiMap.create();
+        userIDs = new HashMap<String, Long>();
+        usersPreferences = new FastByIDMap<PreferenceArray>();
+        userCount = 0;
+        productCount = 0;
+        reviewCount = 0;
+
+        getPreference(filePath);
+        buildRecomender();
+
+    }
+
+    GenericPreference getPreference(String filePath) {
```

Comment:
* Handle access modifier 
    - Investigated what kind of access do you have there.